iaso/models/base.py

```diff
@@ -885,6 +888,37 @@ class Instance(models.Model):
     # See public_create_url workflow in enketo/README.md. used to tell we should export immediately
     to_export = models.BooleanField(default=False)
 
+    def __str__(self):
+        return "%s %s" % (self.form, self.name)
+
+    @property
+    def is_instance_of_reference_form(self) -> bool:
+        if not self.org_unit or not self.org_unit.org_unit_type:
+            return False
+        return self.org_unit.org_unit_type.reference_forms.filter(id=self.form_id).exists()
+
+    @property
+    def is_reference_instance(self) -> bool:
+        if not self.org_unit:
+            return False
+        return self.org_unit.reference_instances.filter(orgunitreferenceinstance__instance=self).exists()
```

Comment:
I'm a bit worried about making  n sql requests if serializing n instances when using this. Has this been checked?

Reply:
Yup, this is why I'm using `prefetch_related` in various Serializers.